src/pages/Sign/sections/SignUp.tsx

```diff
@@ -137,62 +196,216 @@ export const SignUp: React.SFC = () => (
+                                    ) {
+                                      switch (dataStatus.code) {
+                                        case 'started':
+                                          return (
+                                            <SigningStatusText>
+                                              <TranslationsConsumer textKey="SIGN_BANKID_CODE_STARTED">
+                                                {(message) => message}
+                                              </TranslationsConsumer>
+                                            </SigningStatusText>
+                                          )
+                                        case 'userSign':
+                                          return (
+                                            <SigningStatusText>
+                                              <TranslationsConsumer textKey="SIGN_BANKID_CODE_USER_SIGN">
+                                                {(message) => message}
+                                              </TranslationsConsumer>
+                                            </SigningStatusText>
+                                          )
+                                        case 'noClient':
+                                          return (
+                                            <SigningStatusText>
+                                              <TranslationsConsumer textKey="SIGN_BANKID_CODE_NO_CLIENT">
+                                                {(message) => message}
+                                              </TranslationsConsumer>
+                                            </SigningStatusText>
+                                          )
+                                        case 'outstandingTransaction':
+                                          return (
+                                            <SigningStatusText>
+                                              <TranslationsConsumer textKey="SIGN_BANKID_CODE_OUTSTANDING_TRANSACTION">
+                                                {(message) => message}
+                                              </TranslationsConsumer>
+                                            </SigningStatusText>
+                                          )
+                                        default:
+                                          return null
+                                      }
+                                    } else {
+                                      return null
+                                    }
+                                  case SIGNSTATE.COMPLETED: // TODO: backend sets this to null atm
+                                    if (
+                                      dataStatus.status ===
+                                      BANKIDSTATUS.COMPLETE
+                                    ) {
+                                      window.location.href = '/download'
```

Comment:
this should probably be a `Redirect` component from `react-router-dom` so we don't loose what we have in memory